app/dockerdwrapper.c

```diff
@@ -411,33 +433,41 @@ start_dockerd(const struct settings *settings)
 
   g_strlcpy(msg, "Starting dockerd", msg_len);
 
+  if (!use_ipc_socket && !use_tcp_socket) {
+    syslog(LOG_ERR,
+           "At least one of IPC socket or TCP socket must be set to \"yes\". "
+           "dockerd will not be started.");
+    goto end;
+  }
+
   if (use_tls) {
     const char *ca_path = "/usr/local/packages/dockerdwrapper/ca.pem";
     const char *cert_path =
         "/usr/local/packages/dockerdwrapper/server-cert.pem";
     const char *key_path = "/usr/local/packages/dockerdwrapper/server-key.pem";
-
-    args_offset += g_snprintf(args + args_offset,
-                              args_len - args_offset,
-                              " %s %s %s %s %s %s %s %s",
-                              "-H tcp://0.0.0.0:2376",
-                              "--tlsverify",
-                              "--tlscacert",
-                              ca_path,
-                              "--tlscert",
-                              cert_path,
-                              "--tlskey",
-                              key_path);
-
-    g_strlcat(msg, " in TLS mode", msg_len);
-  } else {
+    if (use_tcp_socket) {
```

Comment:
The 3 pem variables above makes no sense if `use_tcp_socket` is false. They were used before, but now they are dead code in 50% of the cases. Think through all cases in the vicinity of your code change to make sure everything still has the same meaning.

By simply expanding `if (use_tls)` to `if (use_tcp_socket && use_tls)` so it becomes:

```
if (use_tcp_socket && use_tls)
    ...
else (use_tcp_socket && !use_tls)
    ...
```

the code would maintain its original symmetry, which often simplifies reading and is sometimes an indication of that nothing was missed.

However, there is some repetition (`use_tcp_socket` is mentioned twice), so by instead simply putting the new condition outside the original code, you get a change that will look slightly bigger due to the whitespace changes, but is actually smaller and the one that best describes how we treat the parameters - we ignore `use_tls` if `use_tcp_socket` isn't set:

```
if (use_tcp_socket) {
    if (use_tls)
        ...
    else
        ...
}